app/uk/gov/hmrc/gform/email/EmailService.scala

```diff
@@ -20,20 +20,26 @@ import uk.gov.hmrc.gform.sharedmodel.form.{ Form, FormField }
 import cats.implicits._
 import play.api.Logger
 import uk.gov.hmrc.gform.auditing.loggingHelpers
-import uk.gov.hmrc.gform.sharedmodel.formtemplate.EmailParameter
+import uk.gov.hmrc.gform.sharedmodel.formtemplate.{ EmailParameter, EmailParametersRecalculated }
 
 import scala.concurrent.{ ExecutionContext, Future }
 import uk.gov.hmrc.http.HeaderCarrier
 
 class EmailService(emailConnector: EmailConnector) {
-  def sendEmail(optemailAddress: Option[String], templateId: String, parameters: Map[EmailParameter, FormField])(
+  def sendEmail(optemailAddress: Option[String], templateId: String, emailParameters: EmailParametersRecalculated)(
     implicit hc: HeaderCarrier,
     mdc: ExecutionContext): Future[Unit] = {
     Logger.info(s" Sending email, template: $templateId, headers: '${loggingHelpers.cleanHeaderCarrierHeader(hc)}'")
-    optemailAddress.fold(().pure[Future])(email =>
-      emailConnector.sendEmail(new EmailTemplate(Seq(email), templateId, parameters.map {
-        case (key, value) => (key.emailTemplateVariable, value.value)
-      })))
+    optemailAddress.fold(().pure[Future])(
+      email =>
+        emailConnector.sendEmail(new EmailTemplate(
```

Comment:
Consider calling a function to create the EmailTemplate - there's a lot going on in here.